### PR TITLE
Fix for deserialization issue with split scope

### PR DESCRIPTION
**Pull request by aneeshdk**
Fix for deserialization issue with split scope

When deserializing a split scoped function, the flag to indicate the
function is not merged was not updated. This was causing issues when we
try to creat the slots.

### CODE REVIEW

**Comment by aneeshdk** [comment]
@akroshg @tcare Can you please take a look?

File: lib/Runtime/ByteCode/ByteCodeEmitter.cpp [hunk 1]
```diff
@@ -1765,7 +1765,7 @@ void ByteCodeGenerator::InitScopeSlotArray(FuncInfo * funcInfo)
 {
     // Record slots info for ScopeSlots/ScopeObject.
     uint scopeSlotCount = funcInfo->bodyScope->GetScopeSlotCount();
-    uint scopeSlotCountForParamScope = funcInfo->paramScope ? funcInfo->paramScope->GetScopeSlotCount() : 0;
+    uint scopeSlotCountForParamScope = funcInfo->paramScope != nullptr && !funcInfo->paramScope->GetCanMergeWithBodyScope() ? funcInfo->paramScope->GetScopeSlotCount() : 0;
```
Comment (akroshg) [inline lib/Runtime/ByteCode/ByteCodeEmitter.cpp]: why paramScope->GetScopeSlotCount not be 0 when GetCanMergeWithbodyScope is true ? I believe the paramScope->GetScopeSlotCount will be populated only when the scope is split.

**Comment by tcare** [comment]
Do we have a test case for this at all?

File: lib/Runtime/ByteCode/ByteCodeEmitter.cpp (hunk 1 above)
Comment (aneeshdk) [inline lib/Runtime/ByteCode/ByteCodeEmitter.cpp]: Fixed it. Added an assert too.

**Comment by aneeshdk** [comment]
As discussed I ran all the test cases with forceserialized flag and fixed the issues.

**Comment by tcare** [comment]
Looks good. Opened #670 to track periodic testing of other variants including forceserialize.

**Comment by aneeshdk** [comment]
@dotnet-bot test disablejit tests please

**Pull request merged**